### PR TITLE
Initial docker build

### PR DESCRIPTION
**Pull request by arielshaqed**
Tested with `docker buildx .`

### CODE REVIEW

**Comment by arielshaqed** [comment]
@itaiad200 for reviewing the build, @nopcoder for reviewing _and_ FYI for integration.

**Review by nopcoder — APPROVED** [review]
If possible remove the build-base. And add docker ignore / don't copy everything to the container.

**Comment by arielshaqed** [comment]
> If possible remove the build-base.

Not sure I understand.  Are you asking about [multi-stage builds](https://docs.docker.com/develop/develop-images/dockerfile_best-practices/#use-multi-stage-builds)?  Those are recommended, so I intentionally build on a separate image, then copy only the resulting code.  This way I don't include a Go compiler, sources, and strange args and working directories.

Or, are you asking to create a "flat" image with no additional base, i.e. replace
```dockerfile
FROM alpine:3.12.0 AS terminus
```
with
```dockerfile
FROM scratch
```
?  I _don't_ want to that.  It _does_ give an even smaller image, but I find those harder to investigate in production.  (Will do it if you *really* want me to...).

> And add docker ignore / don't copy everything to the container.

Done (I think)!  Just copying the 2 Go source directories into the container now.

PTAL...

**Comment by nopcoder** [comment]
Don't want to use scratch too. what you did is great.
Copy the specific sources or ignoring the rest is cool.

**Review by nopcoder — APPROVED** [review]

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+# Packages required to build
+RUN apk add --no-cache build-base
+
+# Copy project deps first since they don't change often
+COPY go.mod go.sum ./
+RUN go mod download
+
+# Copy project
+COPY ./pkg/ ./pkg/
+COPY ./cmd/ ./cmd/
```
[the remark this answers is not in the document]
Comment (arielshaqed) [reply Dockerfile:16]: No, that doesn't work :stuck_out_tongue_winking_eye: .  [TFM](https://docs.docker.com/engine/reference/builder/#copy) says:

> If <src> is a directory, the entire contents of the directory are copied, including filesystem metadata.
>
> **Note:**
>    The directory itself is not copied, just its contents.

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,30 @@
+FROM golang:1.16.2-alpine AS build
+
+ARG VERSION=dev
+
+WORKDIR /build
+
+# Packages required to build
+RUN apk add --no-cache build-base
```
[the remark this answers is not in the document]
Comment (arielshaqed) [reply Dockerfile:8]: Yeah, bad copy+pasta here.

**Review by arielshaqed — COMMENTED** [review]
Thanks!

This still builds, so cool.

**Pull request merged**